### ISSUE TITLE
Terminating TUI with ctrl+c on Windows causes terminal to act weird

### ISSUE THREAD

**Issue by espenja** [vercel/turborepo#8860]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

https://github.com/espenja/turbo-mouse-scroll-tui

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Windows

### Which canary version will you have in your reproduction?

2.0.9

### Describe the Bug

Running `turbo watch dev` on Windows and cancelling the process with `ctrl+c` does not gracefully shut down the process and causes the terminal to act in weird ways.  

This happens when running `turbo` from global install (shown in video 1 in additional context).  
Running `turbo` from `pnpm` does not seem to cause this "effect" (shown in video 2 in additional context). 

#### What happens after ctrl+c after running global turbo watch dev
* Residual fragments of tui's interface is left on the terminal
* Scrolling with the mouse wheel acts like arrow up/down instead of scrolling the terminal window/buffer

**Disclaimer:** This is reported for 2.0.9 and not the canary release. Discussed this with @Zertsov on Discord and they said to report it anyway.

#### Tested with
- Inside VSCode with pwsh with extensions enabled
- Inside VSCode with pwsh with all extensions disabled
- Outside of VSCode in Windows Terminal with pwsh
- Outside of VSCode in GitBash MINGW64 with xterm

### Expected Behavior

**Expected behavior**: Process exits to previous terminal state without fragments from TUI, and mouse wheel scrolls the terminal buffer/window  
**Actual behavior**: Terminal has lots of fragments from TUI, and mouse wheel acts like arrow up and down and scrolls through terminal command history

### To Reproduce

- Be on Windows 11
- Have turbo version 2.0.9 installed globally
- clone reproduction repo: https://github.com/espenja/turbo-mouse-scroll-tui
- `pnpm install -g turbo`
- `pnpm install`
- `turbo watch dev`
- Cancel process with `ctrl+c`
- Scroll with mouse in terminal
- Observe mouse wheel acting like arrow up/down

##### Failure scenario: Screencap of running turbo as a global install and scrolling with mouse wheel after cancelling process with ctrl+c (clicking image will take you to a YouTube video showing what happens)
[![Running global turbo install with turbo watch dev](https://github.com/user-attachments/assets/f6ec9c8a-fb16-4fb0-ba40-67a94b285b2b)](https://www.youtube.com/watch?v=9sa5KuQZIIY)

##### Working scenario: Screencap of running turbo from pnpm and scrolling with mouse wheel after cancelling process with ctrl+c (clicking image will take you to a YouTube video showing what happens)

[![Running global turbo install with turbo watch dev](https://github.com/user-attachments/assets/72c50cf2-e9fb-4103-9920-c771fbdd3ebe)](https://youtu.be/vbiLEYmnAxo)

##### Possibly related/similar bug reported in tmux repository
https://github.com/tmux/tmux/issues/2148

**Comment by espenja** [comment]
Additional information: After reporting issue #8861, I tested what I reported here in that reproduction branch too (which is a pure reproduction repo created with `npx create-turbo@canary -e with-shell-command`), and I am able to reproduce with `2.0.10-canary.0`

**Comment by WillKirkmanM** [comment]
Hey @espenja, it seems turborepo needs to flush the terminal to remove the fragments similar to how the `clear` command works. In the meantime there are two workarounds you can use:

### Use `CTRL+L` / `clear`
This is to manually flush the terminal of the outputs.

### Use Old Logging Style
Turborepo pushed an update which changed the default TUI from the old one to this new modal / tmux type editor. To go back to the old one, add this flag to your run command

```
--log-order=stream 
```
A full command would look like
```
bunx turbo dev --log-order=stream --filter web --env-mode loose
```
![image](https://github.com/user-attachments/assets/9da47681-6521-43d0-a493-718982dd51fb)

**Comment by espenja** [comment]
While `ctrl+L` clears the terminal, and running the old stream experience is an ok workaround, I think the issue still stands with tui. Look at this example:

https://github.com/user-attachments/assets/9ab6e0a4-9186-419f-8e50-61626c5374cd

After having terminated the turbo process with `ctrl+c` the entire terminal is non-scrollable, even after clearing.
The mouse scroll wheel acts like arrow up/down, the scrollbar for the terminal itself is gone, even the terminal cursor indicator is gone. This doesn't go away until you either completely close the terminal or start another app (in the screen cap I show how I can reset the terminal to the original state by starting vim and closing it again)

My (completely uneducated) guess after having looked inside the code for TUI (I don't know Rust at all) would be that the `ctrl+c` handler in TUI somehow either crashes before it can run the cleanup function, or doesn't run it at all: https://github.com/vercel/turbo/blob/496def6068938f517b19d004032a4a6e02aa6f32/crates/turborepo-ui/src/tui/app.rs#L454

Within this cleanup function, it looks like all the things described in my issue would have been resolved:
- Terminal is cleared
- Mouse capture is disabled
- Raw mode is disabled
- Cursor is set to show

**Comment by lucas-labs** [comment]
In my case, after <kbd>Ctrl</kbd>+<kbd>C</kbd>, the terminal has no colors anymore.

![image](https://github.com/user-attachments/assets/193e4533-cc4b-433f-89ae-0f51ba128a51)

**Comment by espenja** [comment]
I mentioned this in my original description as well, but another workaround is executing turbo from pnpm with `pnpm exec turbo watch lint` instead of global turbo `turbo watch lint`. Running with pnpm and pressing `ctrl+c` seems to exit the process just fine.

**Comment by WillKirkmanM** [comment]
What version is pnpm's turbo?

**Comment by espenja** [comment]
It's the same version as global. I've tested all 2.x major versions, same thing happens with all of them.

**Comment by Facuiguacel** [comment]
Hi, this seems to still be an issue on turbo 2.5.4. In my case the terminal's cursor also dissapears, so I dont know where the character will end up while typing.
The work around in bash is using the `reset` command instead of `clear`. Which is not ideal since it also clears the terminal, not something I always want, but at least it resets the cursor behaviour.

This happens to me on a Bash terminal on Windows 11 with npm.

**Comment by coldino** [comment]
Yup, still happening. This really causes a yucky feeling and stops me recommending turbo to other devs.

**Comment by segevfiner** [comment]
It is likely turborepo is not handling Ctrl-C and reseting the terminal modes. When you do a TUI, you need to handle Ctrl-C (`SIGINT`) which has special emulation on Windows for programs running in a terminal AFAIK. It would also be nice to have a quit key for the TUI, e.g. Press q, or something like that.

**Comment by anthonyshew** [comment]
This was addressed in #11444 and #11487. We should be able to close here. If you're still seeing broken behavior around this, please open a new Issue and we'll be happy to take a look.

**Comment by specialkapa** [comment]
Anyone having this issue on the latest release? I upgrade to 2.10.2 today and I am getting this issue

**Comment by anthonyshew** [comment]
@specialkapa https://github.com/vercel/turborepo/issues/8860#issuecomment-4055039832
